scripts/globals/assault.lua

```diff
@@ -0,0 +1,339 @@
+        return
+    end
+
+    local party = player:getParty()
+
+    if party ~= nil then
+        for _,v in ipairs(party) do
+            if not v:hasKeyItem(ORDERS) or v:getCurrentAssault() ~= assaultID or v:getVar("assaultEntered") ~= 0 or v:getMainLvl() < 50 then
+                player:messageText(player, zones[player:getZoneID()].text.MEMBER_NO_REQS, false)
+                player:instanceEntry(target, 1)
+                return
+            elseif v:getZoneID() ~= player:getZoneID() or v:checkDistance(player) > 50 then
+                player:messageText(player, zones[player:getZoneID()].text.MEMBER_TOO_FAR, false)
+                player:instanceEntry(target, 1)
+                return
+            end
+        end
+    end
+
+    player:createInstance(player:getCurrentAssault(), zoneID)
+end
+
+function assaultUtil.onAssaultonInstanceCreated(player, target, instance, endCS, destinationID)
+    if instance then
+        instance:setLevelCap(player:getVar("AssaultCap"))
+        player:setInstance(instance)
+        player:instanceEntry(target, 4)
+        player:setVar("AssaultCap", 0)
+        player:setVar("Assault_Armband", 1)
+        player:delKeyItem(xi.ki.ASSAULT_ARMBAND)
+        player:setLocalVar("Area", destinationID)
+
+        local party = player:getParty()
+        if (party ~= nil) then
+            for _,v in ipairs(party) do
+                if v:getID() ~= player:getID() and v:getZoneID() == player:getZoneID() then
+                    v:setLocalVar("Area", destinationID)
+                    v:setInstance(instance)
+                    v:startEvent(endCS, destinationID)
+                end
+            end
+        end
+    else
+        player:messageText(player, zones[player:getZoneID()].text.CANNOT_ENTER, false)
+        player:instanceEntry(target, 3)
+    end
+end
+
+function assaultUtil.afterInstanceRegister(player, fireFlies, textTable, mobTable)
+    local instance = player:getInstance()
+    local assaultID = player:getCurrentAssault()
+    local cap = instance:getLevelCap()
+
+    for _, v in pairs(mobTable[assaultID].MOBS_START) do
+        SpawnMob(v, instance)
+    end
+
+    player:messageSpecial(textTable.ASSAULT_START_OFFSET + assaultID, assaultID)
+    player:messageSpecial(textTable.TIME_TO_COMPLETE, instance:getTimeLimit())
+    player:addTempItem(fireFlies)
+    if cap ~= 0 then
+        player:addStatusEffect(xi.effect.LEVEL_RESTRICTION, cap, 0, 0)
+    end
+    if player:getVar("Assault_Armband") == 1 then
+        if cap ~= 0 then
+            for _, v in pairs(mobTable[assaultID].MOBS_START) do
+                if cap == 70 then
+                    cap = 5
+                elseif cap == 60 then
+                    cap = 15
+                elseif cap == 50 then
+                    cap = 25
+                end
+                instance:getEntity(bit.band(v, 0xFFF), xi.objType.MOB):setMobLevel(instance:getEntity(bit.band(v, 0xFFF), xi.objType.MOB):getMainLvl() - cap)
```

Comment:
In all files
* `instance:getEntity(bit.band(FOO, 0xFFF), xi.objType.MOB)` can be replaced with `GetMobByID(FOO, instance)`
* `instance:getEntity(bit.band(FOO, 0xFFF), xi.objType.NPC)` can be replaced with `GetNPCByID(FOO, instance)`